### PR TITLE
test(linters): Enable `testifylint`: `blank-import`, `float-compare`, `negative-positive`, `nil-compare` and `useless-assert`

### PR DESCRIPTION
**Pull request by zak-pawel**
## Checklist

- [x] No AI generated code was used in this PR

## Related issues

resolves https://github.com/influxdata/telegraf/issues/15533
resolves https://github.com/influxdata/telegraf/issues/15534
resolves https://github.com/influxdata/telegraf/issues/15536
resolves https://github.com/influxdata/telegraf/issues/15537
resolves https://github.com/influxdata/telegraf/issues/15538

### CODE REVIEW

**Review by DStrand1 — APPROVED** [review]

File: plugins/inputs/google_cloud_storage/google_cloud_storage_test.go [hunk 1]
```diff
@@ -92,8 +92,8 @@ func TestRunGatherOneItem(t *testing.T) {
 	require.Equal(t, "cpu", metric.Measurement)
 	require.Equal(t, "us-east-1", metric.Tags["tags_datacenter"])
 	require.Equal(t, "localhost", metric.Tags["tags_host"])
-	require.Equal(t, 10.0, metric.Fields["fields_cosine"])
-	require.Equal(t, -1.0975806427415925e-12, metric.Fields["fields_sine"])
+	require.InDelta(t, 10.0, metric.Fields["fields_cosine"], 0.001)
+	require.InDelta(t, -1.0975806427415925e-12, metric.Fields["fields_sine"], 0.001)
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/google_cloud_storage/google_cloud_storage_test.go]: I switched to epsilon here. Also introduced consts.

File: plugins/inputs/statsd/statsd_test.go [hunk 2]
```diff
@@ -1552,7 +1552,8 @@ func TestParse_MeasurementsWithMultipleValues(t *testing.T) {
 	// which adds up to 12 individual datapoints to be cached
 	require.EqualValuesf(t, 12, cachedtiming.fields[defaultFieldName].n, "Expected 12 additions, got %d", cachedtiming.fields[defaultFieldName].n)
 
-	require.EqualValuesf(t, 1, cachedtiming.fields[defaultFieldName].upper, "Expected max input to be 1, got %f", cachedtiming.fields[defaultFieldName].upper)
+	require.InDeltaf(t, 1, cachedtiming.fields[defaultFieldName].upper, 0.001,
+		"Expected max input to be 1, got %f", cachedtiming.fields[defaultFieldName].upper)
```
[the remark this answers is not in the document]
Comment (zak-pawel) [reply plugins/inputs/statsd/statsd_test.go]: Right, stripped.

File: metric/metric_test.go [hunk 3]
```diff
@@ -112,7 +112,7 @@ func TestAddFieldOverwrites(t *testing.T) {
 
 	value, ok := m.GetField("value")
 	require.True(t, ok)
-	require.Equal(t, 42.0, value)
+	require.InDelta(t, 42.0, value, 0.001)
```
Comment (zak-pawel) [inline metric/metric_test.go:115]: I didn't use const here because of circular dependency to/from testutil package.

**Review by srebhan — APPROVED** [review]
Thanks @zak-pawel!

**Pull request merged**